scripts/lib/CIME/namelist.py

```diff
@@ -2047,12 +2105,15 @@ def _parse_namelist_group(self):
             self._settings[group_name] = {}
         self._eat_whitespace()
         while self._curr() != '/':
-            name, values = self._parse_name_and_values()
+            name, values, addto = self._parse_name_and_values()
             dsettings = []
             if self._groupless:
                 if name in self._settings:
                     dsettings = self._settings[name]
-                values = merge_literal_lists(dsettings, values)
+                if not addto:
+                    values = merge_literal_lists(dsettings, values)
+                else:
```

Comment:
Why isn't this block of code indented under the

```python
if name in self._settings
```

conditional? I can imagine that it doesn't matter one way or another for

```python
if not addto:
    values = merge_literal_lists(dsettings, values)
```

but for

```python
else:
    values = self._settings[name] + values
```

I think this will raise an exception if (name in self._settings) is false.

I see that the similar block of code that you added in the following routine is all indented under the `if name in self._settings:` conditional, and that looks more correct to me.